### DISCUSSION TITLE
Using classy-classification on Google Colab, "can't find factory"

### DISCUSSION THREAD

**Discussion by mac389** [explosion/spaCy#12786] [Help: Coding & Implementations]
I can't figure out why Colab doesn't recognize the integration between _spacy_ and _classy-classification_. 

```bash
!python -m spacy download en_core_web_md
!python -m pip install classy-classification
!python -m pip install -U spacy
!python -m pip install spacy-transformers
```

```python 
import spacy, os
import pandas as pd 

# Load the dataset
train = pd.read_csv(os.path.join('.','textcat.train_first.tsv'),sep='\t')
data = {category:train[train['source']==category]['text'].tolist()[:10]
        for category in train['source'].unique()}

print("Training model:")
nlp = spacy.load('en_core_web_md')
nlp.add_pipe("text_categorizer",
             config={"data":data,"model":"spacy"})
print("Trained.",nlp.pipe_names)
nlp.to_disk(os.path.join('.','textcat.train_first'))
```

I can run this code as is with a small data set on my MacBookAir. I need to use Colab to train on a data set with 200k sentences. 

When I upload the code to Colab I get the error: 
>Can't find factory for 'text_categorizer' for language English (en). This usually happens when spaCy calls `nlp.create_pipe` with a custom component name that's not registered on the current language class. If you're using a Transformer, make sure to install 'spacy-transformers'. If you're using a custom component, make sure you've added the decorator `@Language.component` (for function components) or `@Language.factory` (for class components).

>Available factories: attribute_ruler, tok2vec, merge_noun_chunks, merge_entities, merge_subtokens, token_splitter, doc_cleaner, parser, beam_parser, lemmatizer, trainable_lemmatizer, entity_linker, ner, beam_ner, entity_ruler, tagger, morphologizer, senter, sentencizer, textcat, spancat, spancat_singlelabel, future_entity_ruler, span_ruler, textcat_multilabel, en.lemmatizer

**Comment by svlandeg** [comment]
Hi!

As the code works fine on your local machine, I suspect this is an install issue in your Colab environment. You shouldn't need to install `spacy-transformers` by the way - the error code is complaining about `"text_categorizer"` from [`classy-classification`](https://github.com/davidberenstein1957/classy-classification) not being found. This happens when the spaCy installation didn't find that registered function.

Could you double check which versions of `spacy` and `classy-classification` are actually running in your Colab (vs on your home machine)?

**Comment by mac389** [comment]
| Machine | Package | Version |
| ---- | ---- | ---- |
| Home | Spacy | 3.2.6 |
| Home | Classy | No attribute `__version__` |
| Colab | Spacy | 3.5.3 (although error path suggests 3.1) | 
| Colab | Classy | No attribute `__version__`

**Reply by AnaCoda** [reply]
Were you able to resolve this?
